server/middleware/populateJourney.ts

```diff
@@ -0,0 +1,29 @@
+import { RequestHandler } from 'express'
+import { AppointmentSessionDatum } from '../@types/express'
+
+const journeys = ['appointmentJourney', 'bulkAppointmentJourney', 'editAppointmentJourney']
+
+export default function populateJourney(): RequestHandler {
+  return async (req, res, next) => {
+    req.session.appointmentSessionDataMap ??= new Map<string, AppointmentSessionDatum>()
+
```

Comment:
Does this create a new session data map on every user request?

Reply:
No. This is a null coalesce assignment operator where it will only create a session map if the existing one is null or undefined. It will only create a new session map once per user session